Update JEDI hashes (20250407)

RussTreadon-NOAA opened the pull request:
# Description

<!-- Enter PR description here. -->

Weekly update of JEDI hashes

None

# Issues

Resolves #1594

# Automated CI tests to run in Global Workflow
- [x] atm_jjob <!-- JEDI atm single cycle DA !-->
- [x] C96C48_ufs_hybatmDA <!-- JEDI atm cycled DA !-->
- [x] C96C48_hybatmaerosnowDA  <!-- JEDI aero/snow cycled DA !-->
- [x] C48mx500_3DVarAOWCDA <!-- JEDI low-res marine 3DVar cycled DA  !-->
- [x] C48mx500_hybAOWCDA <!-- JEDI marine hybrid envar cycled DA !-->
- [x] C96C48_hybatmDA <!-- GSI atm cycled DA !-->

RussTreadon-NOAA committed:
Update to new stable build on 20250401

RussTreadon-NOAA committed:
Merge branch 'develop' into feature/stable-nightly

RussTreadon-NOAA committed:
Update to new stable build on 20250402

RussTreadon-NOAA committed:
Merge branch 'develop' into feature/stable-nightly

RussTreadon-NOAA committed:
Update to new stable build on 20250403

RussTreadon-NOAA committed:
Merge branch 'develop' into feature/stable-nightly

RussTreadon-NOAA committed:
Update to new stable build on 20250404

RussTreadon-NOAA committed:
update to restore functionality of test_gdasapp_atm_jjob (#1594)

RussTreadon-NOAA committed:
Update to new stable build on 20250405

RussTreadon-NOAA commented:
**WCOSS2 g-w CI**

Install `feature/stable-nightly` at 53276d4 in g-w `develop` at [`9f3615d`](https://github.com/NOAA-EMC/global-workflow/commit/9f3615d6f21f55bd7964d2d1209cb6140e3a78b8) on Dogwood.  Run g-w CI with the following results
```
/lfs/h2/emc/ptmp/russ.treadon/EXPDIR/C48_ATM_stable-nightly
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202103231200        Done    Apr 05 2025 17:30:33    Apr 05 2025 19:05:26
 
/lfs/h2/emc/ptmp/russ.treadon/EXPDIR/C48mx500_3DVarAOWCDA_stable-nightly
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202103241800        Done    Apr 05 2025 17:30:35    Apr 05 2025 17:50:35
202103250000        Done    Apr 05 2025 17:30:35    Apr 05 2025 19:35:25
 
/lfs/h2/emc/ptmp/russ.treadon/EXPDIR/C48mx500_hybAOWCDA_stable-nightly
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202103241800        Done    Apr 05 2025 17:30:37    Apr 05 2025 17:50:40
202103250000        Done    Apr 05 2025 17:30:37    Apr 05 2025 19:05:41
 
/lfs/h2/emc/ptmp/russ.treadon/EXPDIR/C48_S2SWA_gefs_stable-nightly
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202103231200        Done    Apr 05 2025 17:30:40    Apr 05 2025 19:30:42
 
/lfs/h2/emc/ptmp/russ.treadon/EXPDIR/C48_S2SW_stable-nightly
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202103231200        Done    Apr 05 2025 17:30:46    Apr 05 2025 19:05:51
 
/lfs/h2/emc/ptmp/russ.treadon/EXPDIR/C96_atm3DVar_stable-nightly
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202112201800        Done    Apr 05 2025 17:30:48    Apr 05 2025 17:50:55
202112210000        Done    Apr 05 2025 17:30:48    Apr 05 2025 20:05:35
202112210600        Done    Apr 05 2025 17:30:48    Apr 05 2025 19:45:33
 
/lfs/h2/emc/ptmp/russ.treadon/EXPDIR/C96C48_hybatmaerosnowDA_stable-nightly
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202112201200        Done    Apr 05 2025 17:30:50    Apr 05 2025 17:55:59
202112201800        Done    Apr 05 2025 17:30:50    Apr 05 2025 20:00:36
202112210000        Done    Apr 05 2025 17:30:50    Apr 05 2025 19:55:36
 
/lfs/h2/emc/ptmp/russ.treadon/EXPDIR/C96C48_hybatmDA_stable-nightly
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202112201800        Done    Apr 05 2025 17:30:52    Apr 05 2025 17:51:03
202112210000        Done    Apr 05 2025 17:30:52    Apr 05 2025 19:35:42
202112210600        Done    Apr 05 2025 17:30:52    Apr 05 2025 19:45:41
 
/lfs/h2/emc/ptmp/russ.treadon/EXPDIR/C96C48_ufs_hybatmDA_stable-nightly
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202402231800        Done    Apr 05 2025 17:30:54    Apr 05 2025 17:51:08
202402240000        Done    Apr 05 2025 17:30:54    Apr 05 2025 20:30:49
202402240600        Done    Apr 05 2025 17:30:54    Apr 05 2025 20:25:38
 
/lfs/h2/emc/ptmp/russ.treadon/EXPDIR/C96mx100_S2S_stable-nightly
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
199405010000    sfs_fcst_mem000_seg0                    28816198                DEAD                 271         2         110.0
199405010000    sfs_fcst_mem001_seg0                    28816199                DEAD                 271         2         110.0
199405010000    sfs_fcst_mem002_seg0                    28816200                DEAD                 271         2         110.0
```

All g-w CI cases pass except `C96mx100_S2S`.  This is a known failure - see g-w issue [#3531](https://github.com/NOAA-EMC/global-workflow/issues/3531).  The `C96mx100_S2S` case does not exercise GDASApp.

RussTreadon-NOAA commented:
The `gdas_prepoceanobs` failures on Orion and Hercules are for the same reason in the 3DVarAOWCDA and hybAOWCDA g-w CI cases.  Below is the failure from the Orion C48mx500_hybAOWCDA gdas_prepoceanobs
```
+ setpdy.sh[49]'[' -z 20210325 ']'
+ setpdy.sh[53]sed 's/[0-9]\{8\}/20210325/' /work2/noaa/da/role-da/CI/orion/GDASApp/workflow/PR/1595/global-workflow/sorc/gdas.cd/build/gdas/test/gw-ci/../../test/gw-ci/C48mx500_hybAOWCDA/COMROOT/date/t00z
sed: can't read /work2/noaa/da/role-da/CI/orion/GDASApp/workflow/PR/1595/global-workflow/sorc/gdas.cd/build/gdas/test/gw-ci/../../test/gw-ci/C48mx500_hybAOWCDA/COMROOT/date/t00z: No such file or directory
++ setpdy.sh[58]finddate.sh 20210325 s-7
+++ finddate.sh[27]set +x
/apps/contrib/NCEP/libs/hpc-stack/intel-2018.4/prod_util/1.2.2/bin/finddate.sh: line 263: pdstr: unbound variable
+ setpdy.sh[58]DATElne_minus=
+ setpdy.sh[59]export err=1
+ setpdy.sh[59]err=1
+ setpdy.sh[59]err_chk
+ err_chk[14]COREROOT=/gpfs/hps/ptmp/production.core/prepoceanobs.541622
+ err_chk[16]'[' -z 1 ']'
+ err_chk[18]'[' 1 -ne 0 ']'
+ err_chk[21]'[' 1 -eq 139 ']'
+ err_chk[33]err_exit
+ environment[0]set +eux

    -------------------------------------------------------------
    -- FATAL ERROR: Job prepoceanobs.541622 failed RETURN CODE 1
    -- ABNORMAL EXIT at Sat Apr  5 12:04:17 CDT 2025 on orion-06-64
    -------------------------------------------------------------
```

It's not clear to me why this job runs to completion on Hera and Dogwood but fail on Hercules and Orion.

The g-w `develop` used on Hercules, Orion, Hera, and Dogwood is [9f3615d](https://github.com/NOAA-EMC/global-workflow/comhttps://github.com/NOAA-EMC/global-workflow/pull/3515mit/9f3615d6f21f55bd7964d2d1209cb6140e3a78b8).  The hash is associated with g-w PR [#3515](https://github.com/NOAA-EMC/global-workflow/pull/3515).   This PR mentions failures in gdas_prepoceanobs.  It's not clear from the PR how the `setpdy` problem was fixed.

Tagging @DavidHuber-NOAA , @guillaumevernieres

RussTreadon-NOAA commented:
@DavidHuber-NOAA , looks like [we are seeing](https://github.com/NOAA-EMC/GDASApp/pull/1595#issuecomment-2781148407) the same `setpdy` issue [you mention](https://github.com/NOAA-EMC/global-workflow/pull/3515#issuecomment-2778616985) in g-w PR [#3515](https://github.com/NOAA-EMC/global-workflow/pull/3515).  How did you resolve the problem?

RussTreadon-NOAA commented:
This PR is ready for review.  

All GDASApp ctests pass on Hera and WCOSS2 (Dogwood).  The Hercules and Orion failures in 3DVarAOWCDA and hybAOWCDA  gdas_prepoceanobs appear to be g-w issues, not GDASApp issues.

DavidHuber-NOAA commented:
@RussTreadon-NOAA I did not resolve this issue at the time I first discovered it as I thought it may have been related to a file system problem I was seeing at the time.  I have looked into it some more, and I think it is a module file problem in the GDASApp.  This is indicated by this line:

```
/apps/contrib/NCEP/libs/hpc-stack/intel-2018.4/prod_util/1.2.2/bin/finddate.sh: line 263: pdstr: unbound variable
```

Somehow, the old hpc-stack version of prod-util is being used.  I think a module file update is in order.  A quick fix would be to add a `set +eu` … `set_strict` wrapper around all GDASApp jobs’ calls to `setpdy.sh`.

RussTreadon-NOAA committed:
remove hardwired prod_util path, load prod_util module (#1594)

RussTreadon-NOAA commented:
Thank you, @DavidHuber-NOAA !  You are right.  Both the hercules and orion GDAS modulefiles contain
```
prepend_path("PATH","/apps/contrib/NCEP/libs/hpc-stack/intel-2018.4/prod_util/1.2.2/bin")
```
I don't recall the reason for including this line.  I removed the above and replaced it with
```
load("prod_util/2.1.1")
```

The failed gdas_prepoceanobs job successfully ran to completion on both Hercules and Orion.   A hardwired prod_util line is also in the Hera modulefile but it is commented out.  

I removed the hardwired prod_util lines from the Hera, Hercules, and Orion modulefiles.  A load for `prod_util/2.1.1` was added to the Hercules and Orion modulefiles.  Done at 62c3995.

guillaumevernieres reviewed (approved):
:+1:

DavidHuber-NOAA reviewed (approved):
Thanks for finding the fix @RussTreadon-NOAA!

RussTreadon-NOAA commented:
Once this PR is merged into GDASApp `develop`, we can close g-w issue [#3555](https://github.com/NOAA-EMC/global-workflow/issues/3555)

DavidHuber-NOAA commented:
@RussTreadon-NOAA I think we will also need to update the GDASApp hash in the global workflow, correct?

RussTreadon-NOAA commented:
True, I'm thinking from a GDASApp-centric perspective.  GDASApp PR [#1595](https://github.com/NOAA-EMC/GDASApp/pull/1595) can run g-w CI but g-w can not run GDASApp without an updated `sorc/gdas.cd` hash.

RussTreadon-NOAA commented:
GDASApp tests pass on Hera, Hercules, and Orion.  g-w CI passes on Dogwood.  Changes reviewed and approved.  Merge into `develop`.
